### ISSUE TITLE
Add compiler optimization flags to ceed.pc

### ISSUE THREAD

**Issue by jrwrigh** [CEED/libCEED#1658]
See [HONEE thread](https://gitlab.com/phypid/honee/-/merge_requests/50#note_2096632669) for background. 

The idea is that downstream users of libCEED can/should inherit the compilation flags of upstream as they should respond similarly to the same optimization flags.

To do that, we could add a `cflags_extra` key to `ceed.pc` to store then, similar to how PETSc does it. And in that we'd just store the `OPT` variable from the build process.

**Comment by jedbrown** [comment]
I'm fine with providing this key, though it may be tricky to record the right one. For example, when PETSc `--download-libceed`, it sets `CFLAGS` directly, not via the `OPT` convenience (whose sole purpose is to make it easy to give optimization flags for C, C++, and Fortran in one shot (only works if they have compatible vendors). https://gitlab.com/petsc/petsc/-/blob/main/config/BuildSystem/config/packages/libceed.py?ref_type=heads#L34

**Comment by jrwrigh** [comment]
Good point. I'm fine with it being `CFLAGS` instead of `OPT`. The thought behind `OPT` is that it just trims down on the intake of flags. `CFLAGS` seems like the better one for robustness sake.

**Comment by jeremylt** [comment]
Related issue #1732?

**Comment by jrwrigh** [comment]
Yeah, similar, though that issue is for runtime while this is for compile time.
